tests/test_shuffle.py

```diff
@@ -1,12 +1,70 @@
+import uuid
+
 import dask
+import dask.dataframe as dd
+import pytest
+
+
+@pytest.fixture(scope="session")
+def s3_stability_url(s3, s3_bucket):
+    # Ensure that the stability-scratch directory exists,
+    # but do NOT reomove it as multiple test runs could be
+    # accessing it at the same time
+    stability_url = f"{s3_bucket}/stability-scratch"
+    s3.mkdirs(stability_url, exist_ok=True)
+    return stability_url
+
+
+@pytest.fixture(scope="session")
+def s3_test_data_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    test_url = f"{s3_stability_url}/test-data-{uuid.uuid4().hex}"
+
+    try:
+        s3.makedirs(test_url)
+        yield test_url
+    finally:
+        s3.rm(test_url, recursive=True)
+
 
+@pytest.fixture
+def s3_write_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    write_url = f"{s3_stability_url}/test-output-{uuid.uuid4().hex}"
 
-def test_shuffle_simple(small_client):
+    try:
+        s3.makedirs(write_url)
+        yield write_url
+    finally:
+        s3.rm(write_url, recursive=True)
+
+
+@pytest.fixture
+def shuffle_dataset(small_client, s3_test_data_url, s3_storage_options):
```

Comment:
This feels a bit early to me. Can we just inline this down in `test_shuffle_simple` for now? We can always factor this out into a fixture later if we need to, but I'd like to wait until there's a need before doing so

Reply:
done